runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -490,10 +490,14 @@ ClientSessionData::destroyJ9SharedClassCacheDescriptorList()
 
 
 void
-ClientSessionData::clearCaches()
+ClientSessionData::clearCaches(bool locked)
    {
-   TR_ASSERT(!_inUse || _sequencingMonitor->owned_by_self(), "Must have sequencing monitor");
-   TR_ASSERT(_numActiveThreads == 0, "Must have no active threads");
+   if (!locked)
+      {
+      // If accessing without the locks, then no threads must be using this client session
+      TR_ASSERT(!_inUse || _sequencingMonitor->owned_by_self(), "Must have sequencing monitor");
```

Comment:
I think we can leave the first assert outside this `if (locked)` because we are requesting this `clearCaches` method be executed under the sequencing monitor.
The second assert could become:
`TR_ASSERT(_numActiveThreads == 0 || locked, "Must have no active threads when accessing without locks");`